backend/lib/endpoints/schema/posts.js

```diff
@@ -1,53 +1,94 @@
 const S = require("fluent-schema");
 const { strictSchema } = require("./utils");
 
+const { join } = require("path");
+const PATH_TO_SCHEMAS = join(__dirname, "../../models/schemas/v2");
+const { schema: postSchema } = require(join(PATH_TO_SCHEMAS, "post"));
+const { schema: orgSchema } = require(join(PATH_TO_SCHEMAS, "individualUser"));
+const { schema: userSchema } = require(join(
+  PATH_TO_SCHEMAS,
+  "organizationUser",
+));
+
+const EXPIRATION_OPTIONS = ["day", "week", "month", "forever"];
+const POST_OBJECTIVES = postSchema.tree.objective.enum;
+const POST_TYPES = postSchema.tree.types.enum;
+const USER_TYPES = [userSchema.tree.type.enum, orgSchema.tree.type.enum];
+const VISIBILITY_OPTIONS = postSchema.tree.visibility.enum;
+
 const getPostsSchema = {
   querystring: strictSchema()
-    .prop("authorId", S.string())
+    .prop(
+      "filter",
+      strictSchema()
+        .prop(
+          "author.location.coordinates",
+          S.array().items(S.number()).minItems(2).maxItems(2),
+        )
+        .prop("author.type", S.string().enum(USER_TYPES))
+        .prop("types", S.array().items(S.string().enum(POST_TYPES)))
+        .prop("objective", S.string().enum(POST_OBJECTIVES)),
+    )
+    .prop("limit", S.integer())
     .prop("skip", S.integer())
-    .prop("limit", S.integer()),
+    .prop("userId", S.string()),
 };
 
 const createPostSchema = {
   body: strictSchema()
+    .prop("content", S.string().required())
+    .prop("expireAt", S.string().enum(EXPIRATION_OPTIONS).required())
+    .prop(
+      "externalLinks",
+      S.object()
+        .prop("appStore", S.string().format("url"))
+        .prop("email", S.string().format("email"))
+        .prop("playStore", S.string().format("url"))
+        .prop("website", S.string().format("url")),
+    )
+    .prop("language", S.array().items(S.string()))
+    .prop("objective", S.string().enum(POST_OBJECTIVES).required().required())
```

Comment:
I think one `.required()` is enough?

Reply:
Sure, only once is enough, just a typo 🤣

Reply:
LOL I was re-reading it over and over again wondering why there were 2 of them 😅